lib/Client/ReportRequest.php

```diff
@@ -63,9 +57,36 @@ public function toThrift() {
             'runtime'         => $this->_runtime->toThrift(),
             'oldest_micros'   => $this->_reportStartTime,
             'youngest_micros' => $this->_now,
-            'log_records'     => $thriftLogs,
             'span_records'    => $thriftSpans,
             'counters'        => $thriftCounters,
         ]);
     }
+
+    /**
+     * @param Auth $auth
+     * @return ProtoReportRequest A Proto representation of this object.
+     */
+    public function toProto($auth) {
+        $counts = [];
+        foreach ($this->_counters as $key => $value) {
+            $count = new MetricsSample();
+            $count->setName(strval($key));
```

Comment:
I'm assuming the answer is no, but is there not a constructor that just takes those values? PHP is weirdly slow at functional calls, so if possible it's best to avoid them in big loops. I'm also assuming this isn't actually a big loop, so probably no big deal. Also apologies if I'm restating information you already know.

Reply:
I didn't see it at first, due to my lack of PHP understanding, but there's a way to do this w/o function calls, so I will do that.